common/discovery-config.cpp

```diff
@@ -15,21 +15,56 @@ fty::Expected<void> ConfigDiscovery::load(const std::string& path)
     return pack::yaml::deserializeFile(path, *this);
 }
 
+std::ostream& operator<<(std::ostream& ss, ConfigDiscovery::Protocol::Type value){
+    using Type = ConfigDiscovery::Protocol::Type;
+
+    ss << [&]() {
+        switch (value) {
+            case Type::POWERCOM:
+                return "nut_powercom";
+            case Type::XML_PDC:
+                return "nut_xml_pdc";
+            case Type::SNMP:
+                return "nut_snmp";
+            case Type::UNKNOWN:
+                return "unknown";
+        }
+        return "unknown";
+    }();
+    return ss;
+}
+std::istream& operator>>(std::istream& ss, ConfigDiscovery::Protocol::Type& value){
+    using Type = ConfigDiscovery::Protocol::Type;
+
+    std::string strval;
+    ss >> strval;
+    if (strval == "nut_powercom") {
+        value = Type::POWERCOM;
+    } else if (strval == "nut_xml_pdc") {
+        value = Type::XML_PDC;
+    } else if (strval == "nut_snmp") {
+        value = Type::SNMP;
+    } else {
+        value = Type::UNKNOWN;
+    }
+    return ss;
+}
+
 std::ostream& operator<<(std::ostream& ss, ConfigDiscovery::Discovery::Type value)
 {
     using Type = ConfigDiscovery::Discovery::Type;
 
     ss << [&]() {
         switch (value) {
-            case Type::Local:
+            case Type::LOCAL:
```

Comment:
Not sure if it's needed anymore to support all the type as we can done only 2 different scan:

- local (automatic from UI)
- fullscan (ip & range)